blockchain/blockchain.go

```diff
@@ -1397,6 +1415,41 @@ func (bc *BlockChain) gcCachedNodeLoop() {
 	}()
 }
 
+func (bc *BlockChain) pruneTrieNodeLoop() {
+	// ReadPruningMarks(1, limit) is very slow because it iterates over the most of MiscDB.
+	// ReadPruningMarks(start, limit) is much faster because it only iterates a small range.
+	startNum := uint64(1)
+
+	bc.wg.Add(1)
+	go func() {
+		defer bc.wg.Done()
+		for {
+			select {
+			case num := <-bc.chPrune:
+				if !bc.db.ReadPruningEnabled() || bc.cacheConfig.LivePruningRetention == 0 {
+					continue
+				}
+				if num <= bc.cacheConfig.LivePruningRetention {
+					continue
+				}
+				limit := num - bc.cacheConfig.LivePruningRetention // Prune [1, latest - retention]
+
+				startTime := time.Now()
+				marks := bc.db.ReadPruningMarks(startNum, limit+1)
+				bc.db.PruneTrieNodes(marks)
+				bc.db.DeletePruningMarks(marks)
+
+				logger.Info("Pruned trie nodes", "number", num, "start", startNum, "limit", limit,
+					"count", len(marks), "elapsed", time.Since(startTime))
```

Comment:
```suggestion
				logger.Info("Pruned trie nodes", "from", startNum, "to", limit,
					"count", len(marks), "elapsed", time.Since(startTime))
```
Does it feel more informative to you? (Additionally, the `number` seems not informative in this log)

Reply:
I left it because "Pruned trie nodes" log is printed out-of-order with respect to block insertion logs.

```
INFO[07/06,07:48:08 Z] [5] Inserted a new block                      number=9999871 hash=d2089b…c99ab6 txs=0     gas=0          elapsed=2.526ms      processTxs=1.307µs   finalize=712.298µs validateState=69.476µs  totalWrite=1.203515ms   trieWrite=983.898
INFO[07/06,07:48:09 Z] [49] Persisted trie from memory database       blockNum=9999872 updated nodes=6003   updated nodes size=2.64mB   time=65.080295ms  gcnodes=4632   gcsize=2.40mB   gctime=14.170282ms  livenodes=4735   livesize=2.58mB   preimages=1051
INFO[07/06,07:48:09 Z] [5] Inserted a new block                      number=9999872 hash=fbdef9…9208da txs=1     gas=57117      elapsed=69.023ms     processTxs=344.698µs finalize=987.294µs validateState=95.961µs  totalWrite=67.023094ms  trieWrite=66.7293
INFO[07/06,07:48:09 Z] [5] Inserted a new block                      number=9999873 hash=37990f…af3163 txs=2     gas=788464     elapsed=8.285ms      processTxs=2.123ms   finalize=2.001ms   validateState=121.781µs totalWrite=3.292306ms   trieWrite=2.84218
INFO[07/06,07:48:09 Z] [5] Inserted a new block                      number=9999874 hash=70ac3a…31be14 txs=1     gas=741619     elapsed=7.836ms      processTxs=2.290ms   finalize=1.815ms   validateState=111.841µs totalWrite=2.736272ms   trieWrite=2.46042
INFO[07/06,07:48:09 Z] [5] Pruned trie nodes                         number=9999872 start=9826945 limit=9827072 count=12894  elapsed=30.478143ms
INFO[07/06,07:48:09 Z] [5] Inserted a new block                      number=9999875 hash=065177…63dc39 txs=1     gas=57117      elapsed=12.240ms     processTxs=442.191µs finalize=1.295ms   validateState=106.877µs totalWrite=9.544038ms   trieWrite=1.26109
INFO[07/06,07:48:09 Z] [5] Inserted a new block                      number=9999876 hash=6251db…fa4e69 txs=0     gas=0          elapsed=2.754ms      processTxs=1.341µs   finalize=812.607µs validateState=69.993µs  totalWrite=1.271557ms   trieWrite=1.02655
INFO[07/06,07:48:09 Z] [5] Inserted a new block                      number=9999877 hash=bfafae…2820de txs=2     gas=114192     elapsed=4.168ms      processTxs=720.532µs finalize=1.129ms   validateState=100.133µs totalWrite=1.527832ms   trieWrite=1.29307

```